lib/chef/resource/chef_client_launchd.rb

```diff
@@ -110,21 +110,51 @@ class ChefClientLaunchd < Chef::Resource
           end
         end
 
-        launchd "com.chef.chef-client" do
+        launchd "com.#{Chef::Dist::SHORT}.#{Chef::Dist::CLIENT}" do
           username new_resource.user
           working_directory new_resource.working_directory
           start_interval new_resource.interval * 60
           program_arguments ["/bin/bash", "-c", client_command]
           environment_variables new_resource.environment unless new_resource.environment.empty?
           nice new_resource.nice
           low_priority_io true
-          action :enable
+          notifies :sleep, "chef_sleep[Sleep before client restart]", :immediately
+          action :create # create only creates the file. No service restart triggering
+        end
+
+        # Launchd doesn't have the concept of a reload aka restart. Instead to update a daemon config you have
+        # to unload it and then reload the new plist. That's usually fine, but not if chef-client is trying
+        # to restart itself. If the chef-client process uses launchd or macosx_service resources to restart itself
+        # we'll end up with a stopped service that will never get started back up. Instead we use this daemon
+        # that triggers when the chef-client plist file is updated, and handles the restart outside the run.
+        launchd "com.#{Chef::Dist::SHORT}.restarter" do
+          username "root"
+          watch_paths ["/Library/LaunchDaemons/com.#{Chef::Dist::SHORT}.#{Chef::Dist::CLIENT}.plist"]
+          standard_out_path "#{::File.join(new_resource.log_directory, new_resource.log_file_name)}"
+          standard_error_path "#{::File.join(new_resource.log_directory, new_resource.log_file_name)}"
```

Comment:
```suggestion
          standard_out_path ::File.join(new_resource.log_directory, new_resource.log_file_name)
          standard_error_path ::File.join(new_resource.log_directory, new_resource.log_file_name)
```

These shouldn't need interpolation.

Reply:
Yep. We should turn on that RuboCop cop. I found a few in Ohai the other day using it.

Reply:
oh yeah, or that.  did chefstyle not complain about useless interpolation?

Reply:
https://github.com/chef/chef/pull/10394

Reply:
We'd need to turn on Style/RedundantInterpolation to get this everywhere